providers/provider-protocol-reference.md

```diff
@@ -0,0 +1,202 @@
+
+All functions decribed here take msgpack-serialized payloads as arguments and
+return values. The key/values to use for these payloads are exactly the ones you
+can find in `fiberplane::models` crate documentation (NOTE: put the docs.rs link
+once fiberplane is open-sourced and published).
+
+## Imported functions
+
+These functions are usable as long as you import `fiberplane_pdk::prelude::*` in
+your library code. These functions exist to allow you to use OS-level features by
+asking the runtime for information.
+
+For Fiberplane Daemon, these functions are implemented in [Provider Runtime
+crate](https://github.com/fiberplane/fiberplane-rs/blob/main/fiberplane-provider-protocol/fiberplane-provider-runtime/src/spec/mod.rs),
+should you want to inspect the source.
+
+### `make_http_request`
+
+#### Signature
+
+```rust
+pub async fn make_http_request(request: HttpRequest) -> Result<HttpResponse, HttpRequestError>;
+```
+
+#### Purpose
+
+This is the most useful imported method, it allows the providers to make HTTP queries to
+remote hosts through the hosting runtime. A basic example of its usage can be seen in the
+tutorial to create a provider:
+
+```rust
+    let base_url: Url = "https://jsonplaceholder.typicode.com".parse().unwrap();
+    let url = base_url
+        .join("/users")
+        .unwrap();
+
+    let request = HttpRequest {
+        url,
+        headers: None,
+        method: HttpRequestMethod::Get,
+        body: None,
+    };
+
+    let response = make_http_request(request).await?;
+    
+    serde_json::from_slice(&response.body).map_err(|e| Error::Deserialization { message: format!("Could not deserialize payload: {e:?}") })
+```
+
+### `log`
+
+#### Signature
+
+```rust
+pub fn log(message: String);
+```
+
+#### Purpose
+
+This function asks the hosting runtime to log a message. This will make your log
+messages appear in Studio or in the standard output of the Fiberplane Daemon
+instance.
+
+### `random`
+
+#### Signature
+
+```rust
+pub fn random(len: u32) -> Vec<u8>;
+```
+
+#### Purpose
+
+Query the hosting runtime for a source of randomness. Note that there is no
+guarantee that the random values you obtain this way to be cryptographically
+secure: it is entirely dependent on the runtime implementation. You can see the
+implementation of randomness for Fiberplane Daemon in the [Provider Runtime
+crate](https://github.com/fiberplane/fiberplane-rs/blob/6679fbd0f1cbdac7c57422ae699e12bb35bed71b/fiberplane-provider-protocol/fiberplane-provider-runtime/src/spec/mod.rs#L116)
+source code
+
+### `now`
+
+#### Signature
+
+```rust
+pub fn now() -> Timestamp;
+```
+
+#### Purpose
+
+Returns a timestamp of the current time of execution. The usefulness of this
+timestamp comes from what you want to do with timers, or signature
+creation/verification schemes.
+
+
+## Exported functions
+
+It is usually safer to use the Plugin Development Kit macros to let the compiler
+generate correct implementations of these functions
```

Comment:
This does not apply to all exported functions, so I'd remove this mention here and insert a similar disclaimer specifically with the `get_supported_query_types()` and `invoke2()` functions.

Reply:
What does apply is that any function that you wish to export needs to be annotated with `#[pdk_export]`. I would mention this here.